common/buildcraft/factory/tile/TileTank.java

```diff
@@ -63,11 +68,25 @@ public IdAllocator getIdAllocator() {
         return IDS;
     }
 
+    @Override
+    public void readFromNBT(NBTTagCompound nbt) {
+        super.readFromNBT(nbt);
+
+        fluidAmountPrev = tank.getFluidAmount();
+        comparatorLevel = MathUtil.scaledClamp(tank.getFluidAmount(), tank.getCapacity(), 15);
+    }
+
     // ITickable
 
     @Override
     public void update() {
         smoothedTank.tick(getWorld());
+
+        if (tank.getFluidAmount() != fluidAmountPrev) {
+            fluidAmountPrev = tank.getFluidAmount();
+            comparatorLevel = MathUtil.scaledClamp(tank.getFluidAmount(), tank.getCapacity(), 15);
+            markDirty();
+        }
```

Comment:
Perhaps compare with the previous comparator level rather than the previous amount of fluid?